docs/template_merging.md

```diff
@@ -0,0 +1,147 @@
+# Overlaid Definitions
+
+The way BIG-IP FAST generates parameter definitions can be surprising at times if that parameter shows up multiple times in the template text.
+
+When generating parameter definitions, BIG-IP FAST looks at the following locations in the following order, with later definitions overriding/modifying previous ones:
+1. Embedded mustache tags in any merged templates.  For example: {{var:f5:port}}
+2. The definitions properties of any merged templates. Templates are merged by name using $ref inside a oneOf, anyOf, or allOf clause.
+3. Embedded mustache tags in the primary template.
+4. The definitions property in the primary template.
+5. The parameters property in any merged templates.
+6. The parameters property in the primary template.
+
+#### Notes:
+* If a duplicate Mustache tag exists in the template, then the last encountered tag is used for the definition. The order that Mustache tags are parsed in should not be assumed.
+* Properties within the definition (e.g., title, description, type, format, default, etc.) are merged together as they are found with newer data taking precedence over old data on key conflicts.
+* Values from the parameters property of YAML templates will be used in place of the default from the parameter definition but will not actually update the definition itself.
+
+
+# JSON Schema Basic Types
+
+## Definitions
+JSON Schema allows us to define auxiliary schema in order to be reused and combined later on.
+This involves two steps:
+1. We need to define the subschemas to be used later on, and
+2. We need a standard for calling and reusing these definitions.
+
+To establish a difference between the main schema and the auxiliary definitions, we adopt the convention that every JSON Schema document consists of two parts; a JSON Schema, and a set of definitions.
+
+For example, if we want a definition for virtuals, it may look like this:
+```yaml
+definitions:
+    virtuals:
+        type: array
+        items:  {
+                type: string,
+                format: ipv4
+            }
+```
+
+> **See Also:** [JSON Editor: $ref and definitions](https://github.com/json-editor/json-editor#ref-and-definitions) for additional code examples.
+
+**Array:** Arrays are used for ordered elements.
+In JSON, each element in an array may be of a different type. Elements of the array may be ordered or unordered based on the API being templated. This section covers typical JSON schema definitions for common patterns.
```

Comment:
FAST is not just for templating APIs. This also mentions covering typical definitions for common patterns, but then it only shows one example.

Reply:
What is another example of a a typical definitions for common patterns that we can use?

Reply:
Moved to a new JIRA item.